enabler/src/de/schildbach/pte/VrsProvider.java

```diff
@@ -688,6 +686,11 @@ public QueryTripsResult queryTrips(final Location from, final @Nullable Location
                     QueryTripsResult.Status.UNKNOWN_TO);
         }
 
+        if (options == null)
+            options = new TripOptions();
+        if (options.products == null)
```

Comment:
These two lines also seem to be unrelated, but let me know if I'm wrong.

Reply:
Hum they are kind of related, because in line 700 (old file) it expected the `products` to be non-null, so that's what I wanted to ensure here.

Reply:
I understand the problem, and I agree the NPE should be fixed. It's just not related to this PR, as the bug has existed before.

I just fixed the bug in master via 458fea99ac0bbe4dc8a526bb9e98d8c4f3eabb88. If you rebase your PR on master the need for this specific change should go away.